trie/hexary.py

```diff
@@ -171,29 +172,12 @@ def _get_node(self, node_hash):
             encoded_node = node_hash
         else:
             encoded_node = self.db[node_hash]
-        node = self._decode_node(encoded_node)
+        node = decode_node(encoded_node)
 
         return node
 
     def _persist_node(self, node):
-        validate_is_node(node)
-        if is_blank_node(node):
-            return BLANK_NODE
-        encoded_node = rlp.encode(node)
-        if len(encoded_node) < 32:
-            return node
-
-        encoded_node_hash = keccak(encoded_node)
-        self.db[encoded_node_hash] = encoded_node
-        return encoded_node_hash
-
-    def _decode_node(self, encoded_node_or_hash):
-        if encoded_node_or_hash == BLANK_NODE:
-            return BLANK_NODE
-        elif isinstance(encoded_node_or_hash, list):
-            return encoded_node_or_hash
-        else:
-            return rlp.decode(encoded_node_or_hash)
+        return persist_node(self.db, node)
```

Comment:
Seems like just removing the `_persist_node` method and changing all internal uses of `self._persist_node` would be cleaner.

Reply:
I kept it because then callsites don't have to pass `self.db` as the first argument. Don't feel strong about it, though, so could do as you suggest if you do. But just realized that I'm no longer using that function in TrieSync, so I can revert this change.